extensions/amp-video-iframe/0.1/amp-video-iframe.js

```diff
@@ -0,0 +1,382 @@
+/**
+ * Copyright 2018 The AMP HTML Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS-IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+import {Deferred} from '../../../src/utils/promise';
+import {LayoutPriority} from '../../../src/layout';
+import {Services} from '../../../src/services';
+import {VideoEvents} from '../../../src/video-interface';
+import {
+  createFrameFor,
+  isJsonOrObj,
+  objOrParseJson,
+  originMatches,
+} from '../../../src/iframe-video';
+import {dev} from '../../../src/log';
+import {dict} from '../../../src/utils/object';
+import {getData, listen} from '../../../src/event-helper';
+import {htmlFor} from '../../../src/static-template';
+import {
+  installVideoManagerForDoc,
+} from '../../../src/service/video-manager-impl';
+import {isAdLike, looksLikeTrackingIframe} from '../../../src/iframe-helper';
+import {isLayoutSizeDefined} from '../../../src/layout';
+import {removeElement} from '../../../src/dom';
+
+/** @private @const */
+const TAG = 'amp-video-iframe';
+
+/** @private @const */
+const SANDBOX = 'allow-scripts allow-same-origin';
+
+/**
+ * Events allowed to be dispatched from messages.
+ * @private @const
+ */
+const ALLOWED_EVENTS = [
+  VideoEvents.PLAYING,
+  VideoEvents.PAUSE,
+  VideoEvents.ENDED,
+  VideoEvents.MUTED,
+  VideoEvents.UNMUTED,
+  VideoEvents.AD_START,
+  VideoEvents.AD_END,
+];
+
+/** @implements {../../../src/video-interface.VideoInterface} */
+class AmpVideoIframe extends AMP.BaseElement {
+
+  /** @param {!AmpElement} element */
+  constructor(element) {
+    super(element);
+
+    /** @private {boolean} */
+    this.isAdLike_ = false;
+
+    /** @private {boolean} */
+    this.isTrackingIframe_ = false;
+
+    /** @private {?Element} */
+    this.iframe_ = null;
+
+    /** @private {!UnlistenDef|null} */
+    this.unlistenFrame_ = null;
+
+    /** @private {?function()} */
+    this.readyPromise_ = null;
+
+    /** @private {?function()} */
+    this.readyResolver_ = null;
+
+    /** @private {?function()} */
+    this.readyRejecter_ = null;
+
+    /** @private {boolean} */
+    this.canPlay_ = false;
+
+    /**
+     * @param {!Event} e
+     * @private
+     */
+    this.boundOnMessage_ = e => this.onMessage_(e);
+  }
+
+  /** @override */
+  isLayoutSupported(layout) {
+    return isLayoutSizeDefined(layout);
+  }
+
+  /** @override */
+  onLayoutMeasure() {
+    const {element} = this;
+    this.isAdLike_ = isAdLike(element);
```

Comment:
we can move these to `buildCallback` and just throw. we don't need to support these same way amp-iframe does.